### ISSUE TITLE
UnsupportedOperationException in reverse() of JTS-LineStrings created from GeoLatte-LineStrings

### ISSUE THREAD

**Issue by michikommader** [GeoLatte/geolatte-geom#38]
In GeoLatte 1.0.1 I cannot `reverse()` a `com.vividsolutions.jts.geom.LineString` which was created from a `org.geolatte.geom.LineString` through `JTS.to()`-Method. Consider this example:
```
PositionSequenceBuilder<C2D> sequencer = PositionSequenceBuilders.fixedSized(2, C2D.class);
sequencer.add(0, 0);
sequencer.add(1, 0);
PositionSequence<C2D> sequence = sequencer.toPositionSequence();
org.geolatte.geom.LineString gLine = Geometries.mkLineString(sequence, CrsRegistry.getProjectedCoordinateReferenceSystemForEPSG(32632));
com.vividsolutions.jts.geom.LineString jLine = (com.vividsolutions.jts.geom.LineString) JTS.to(gLine);
jLine.reverse();
```
This throws an `java.lang.UnsupportedOperationException` because `org.geolatte.geom.AbstractPositionSequence#setOrdinate()` is not implemented yet. As current workaround you manually have to duplicate the `com.vividsolutions.jts.geom.LineString` through the JTS GeometryFactory in advance.

**Comment by maesenka** [comment]
PositionSequences are immutable,so that is why the setOrdinate method is not implemented, and never will. 

I'll add a method to reverse Geometries in the ProjectedGeometryOperations interface. This is the interface that collects operations defined for geometries in a ProjectedCoordinateSystem. Then you can directly reverse Geolatte.LineStrings

**Comment by michikommader** [comment]
But still I cannot `reverse()` an ordinary JTS-LineString anymore, after it has been created from a GeoLatte-LineString using the the `org.geolatte.geom.jts.JTS.to()`-Method. This complicates things unnecessarily. After morphing the geometry from GeoLatte to JTS the user should not be in need to know that it originated from GeoLatte and thus must be able to use the native JTS `reverse()`-method on this particular JTS geometry without problems.
I have a huge geo-processing framework based on JTS which is partially persisted through Hibernate. After switching from Hibernate 4.x to Hibernate 5.x and thus to persistent GeoLatte geometry types, my processing is still in JTS and "does not know anything about GeoLatte". Being not able to use JTS-`reverse()` directly any more forces me either to do a hard-copy of the `CoordinateSequence` each time, or to use `reverse()` from GeoLatte's `ProjectedGeometryOperations`, which are theoretically detached from my pure JTS processing framework.

**Comment by maesenka** [comment]
You're right: it's not really possible to keep strictly immutable `PositionSequences` and still have complete JTS compatibility. Thank you for pointing this out.

 It's fixed now in commit b0e8fb47c5ca33f020bfcf97a5f22a944b1d24e4.
